backend/tests/integration/pidstore/minters/test_minters_bai.py

```diff
@@ -81,12 +84,182 @@ def test_bai_minter_without_deleting_all_external_pids(inspire_app):
     new_bai = RecordProvider.bai()
     data = dict(rec)
     data["ids"][0]["value"] = new_bai
-    rec.update(data)
+    with override_config(FEATURE_FLAG_ENABLE_BAI_PROVIDER=False):
+        rec.update(data)
     bai_pid = PersistentIdentifier.query.filter_by(pid_type="bai").one()
 
     assert bai_pid.pid_value == new_bai
-
-    rec.delete()
+    with override_config(FEATURE_FLAG_ENABLE_BAI_PROVIDER=False):
+        rec.delete()
     bai_count = PersistentIdentifier.query.filter_by(pid_type="bai").count()
 
     assert bai_count == 0
+
+
+def test_minter_bai_provided(inspire_app):
+    data = {
+        "ids": [
+            {"schema": "JACOW", "value": "JACoW-12345678"},
+            {"schema": "INSPIRE BAI", "value": "K.Janeway.1"},
+        ]
+    }
+    expected_pids_len = 1
+    epxected_pids_values = [data["ids"][1]["value"]]
+    expected_pids_provider = "bai"
+    expected_pids_status = PIDStatus.REGISTERED
+    with override_config(FEATURE_FLAG_ENABLE_BAI_PROVIDER=True):
+        record = create_record("aut", data=data)
+    result_pids = (
+        PersistentIdentifier.query.filter_by(object_uuid=record.id)
+        .filter_by(pid_type="bai")
+        .all()
+    )
+    result_pids_len = len(result_pids)
+
+    assert expected_pids_len == result_pids_len
+
+    pid = result_pids[0]
+    assert expected_pids_provider == pid.pid_provider
+    assert expected_pids_status == pid.status
+    assert pid.pid_value in epxected_pids_values
+
+
+def test_minter_bai_new(inspire_app):
+    with override_config(FEATURE_FLAG_ENABLE_BAI_PROVIDER=True):
+        record = create_record("aut")
+
+    expected_pids_len = 1
+
+    result_pids_len = PersistentIdentifier.query.filter_by(
+        object_uuid=record.id, status=PIDStatus.REGISTERED, pid_type="bai"
+    ).count()
+
+    assert expected_pids_len == result_pids_len
+    bai_entry = record["ids"][0]
+    assert bai_entry["schema"] == "INSPIRE BAI"
```

Comment:
why not assert the expected value? This is very generic, we don't test if the actual BAI

Reply:
There are tests with static name in author in which generated bai is checked/validated. Here random name is used so it's pointless to check if it's correct, as I would have to add logic for generating it in this test or use logic from BAI which is tested already in other place.

Reply:
then isn't it this test redundant? I don't understand what it tests just the bai schema?

Reply:
> What is the plan with the existing BAIs what we'll do with them when we enable this?

Existing BAIs should be used, only change is that new PID will be generated if record is missing BAI.